// hello-world/src/try/composition_n_inheritance.js
import React, {Component} from 'react';

// M1:
function FancyBorder(props) {
  return (
    <div className={'FancyBorder FancyBorder-' + props.color} >
      {props.children}
    </div>
  );
}

function WelcomeDialog() {
  return (
    <FancyBorder color="blue">
      <h1 className="Dialog-title">
        Welcome
      </h1>
      <p className="Dialog-message">
        Thank you for visiting our spacecraft!
      </p>
    </FancyBorder>
  );
}

// M2:
function SplitPane(props) {
  return (
    <div className="SplitPane">
      <div className="SplitPane-left">
        {props.left}
      </div>
      <div className="SplitPane-right">
        {props.right}
      </div>
    </div>
  );
}

class Contacts extends Component{
  render() {
    return <div>im left</div>;
  }
}
class Chat extends Component{
  render() {
    return <div>im right</div>;
  }
}

function App() {
  return (
    <SplitPane
      left={
        <Contacts />
      }
      right={
        <Chat />
      }
    />
  )
}

export default App;
