// hello-world/src/try/forms.js
import React, {Component} from 'react';

// class Forms extends Component {
//   // constructor(props){
//   //   super(props);
//   // }
// 
//   render() {
//     return (
//       <form onSubmit={this.handleSubmit}>
//         <Input /><br/> 
//         {/* <Textarea /> */}
//         <input type="submit" value="Submit" />
//       </form>
//     );
//   }
// }

/* Controlled Component */
// input
//class Forms1 extends Component {
//  constructor(props){
//    super(props);
//    this.state = {value: ''};
//    
//    this.handleChange = this.handleChange.bind(this);
//    this.handleSubmit = this.handleSubmit.bind(this);
//  }
//
//  handleChange(event) {
//    this.setState({value: event.target.value});
//  }
//
//  handleSubmit(event) {
//    alert('A name was submitted: ' + this.state.value);
//    alert('An essay was submitted: ' + this.state.value);
//    event.preventDefault();
//  }
//
//  render (){
//    return (
//      <form onSubmit={this.handleSubmit}>
//        <label>
//          Name:
//          <input type="text" value={this.state.value} onChange={this.handleChange} />
//        </label>
//        <input type="submit" value="Submit" />
//      </form>
//    );
//  }
//}
//
//// textarea
//class Forms2 extends Component {
//  constructor(props) {
//    super(props);
//    this.state = {
//      value: 'Please write an essay about your favorite Dom element.'
//    };
//
//    this.handleChange = this.handleChange.bind(this);
//    this.handleSubmit = this.handleSubmit.bind(this);
//  }
//
//  handleChange(event) {
//    this.setState({value: event.target.value});
//  }
//
//  handleSubmit(event) {
//    alert('An essay was submitted: ' + this.state.value);
//    event.preventDefault();
//  }
//
//  render() {
//    return (
//      <form onSubmit={this.handleSubmit}>
//        <label>
//          Essay:
//          <textarea value={this.state.value} onChange={this.handleChange} />
//        </label>
//        <input type="submit" value="Submit" />
//      </form>
//    );
//  }
//}
//
//// select
//class Forms3 extends Component {
//  constructor(props) {
//    super(props);
//    this.state = {value: 'coconut'}; // default selected
//
//    this.handleChange = this.handleChange.bind(this);
//    this.handleSubmit = this.handleSubmit.bind(this);
//  }
//
//  handleChange(event) {
//    this.setState({value: event.target.value});
//  }
//
//  handleSubmit(event) {
//    alert('Your favorite flavor is: ' + this.state.value)
//    event.preventDefault();
//  }
//
//  render() {
//    return (
//      <form onSubmit={this.handleSubmit}>
//        <label>
//          Pick your favorite La Croix flavor:
//          <select value={this.state.value} onChange={this.handleChange}>
//            <option value="grapefruit">Grapefruit</option>
//            <option value="lime">Lime</option>
//            <option value="coconut">Coconut</option>
//            <option value="mango">Mango</option>
//          </select>
//        </label>
//        <input type="submit" value="submit" />
//      </form>
//    );
//  }
//}

// multi Inputs
class Forms extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isGoing: true,
      numberOfGuests: 2
    };

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  render() {
    return (
      <form>
        <label>
          Is going:
          <input
            name="isGoing"
            type="checkbox"
            checked={this.state.isGoing}
            onChange={this.handleInputChange} />
        </label>
        <br />
        <label>
          Number of guests:
          <input
            name="numberOfGuests"
            type="number"
            value={this.state.numberOfGuests}
            onChange={this.handleInputChange} />
        </label>
      </form>
    );
  }
}

export default Forms;
